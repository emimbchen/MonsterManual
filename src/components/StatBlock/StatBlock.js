function StatBlock({ label, value, ...props }) {
  return (
    <div {...props}>
      {label && <div className="font-bold uppercase">{label}</div>}
      {value && <span className="capitalize">{value}</span>}
    </div>
  );
}

export default StatBlock;
