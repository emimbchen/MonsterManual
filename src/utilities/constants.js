export const SPELL_LEVEL_COLORS = {
  0: "#B3E5FC",
  1: "#81C784",
  2: "#FFEB3B",
  3: "#FFB74D",
  4: "#F06292",
  5: "#BA68C8",
  6: "#9575CD",
  7: "#7986CB",
  8: "#64B5F6",
  9: "#4DD0E1",
};
