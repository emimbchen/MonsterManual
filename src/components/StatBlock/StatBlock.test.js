import axe from "axe-core";
import StatBlock from "./StatBlock";
import { render, screen } from "@testing-library/react";
import { MOCK_STAT } from "../../utilities/mockData";

const mockStatBlock = (
  <StatBlock label={MOCK_STAT.label} value={MOCK_STAT.value} />
);

describe("StatBlock Component", () => {
  test("Accessibility check", async () => {
    const { container } = render(mockStatBlock);
    const results = await axe.run(container);
    expect(results.violations.length).toBe(0);
  });

  it("renders correctly", () => {
    const { asFragment } = render(mockStatBlock);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays values", () => {
    render(mockStatBlock);
    const label = screen.getByText(MOCK_STAT.label);
    const value = screen.getByText(MOCK_STAT.value);
    expect(label).toBeInTheDocument();
    expect(value).toBeInTheDocument();
  });

  it("handles empty input", () => {
    const { container } = render(<StatBlock />);
    expect(container.firstChild).toBeEmptyDOMElement();
  });
});
