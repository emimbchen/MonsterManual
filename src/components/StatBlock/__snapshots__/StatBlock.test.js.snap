// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatBlock Component renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="font-bold uppercase"
    >
      This is a label
    </div>
    <span
      class="capitalize"
    >
      This is the data
    </span>
  </div>
</DocumentFragment>
`;
